Bottom: f8bb5b18c7c26fe25684a50bbbdcbba2ea9a11ea
Top:    6099427d75b28c69a109f129329f56c7c744d864
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-10-16 15:54:35 +0100

Refresh of add-support-for-depth-to-trian

---

diff --git a/src/hid/common/hidgl.h b/src/hid/common/hidgl.h
index cf38e16..7c71398 100644
--- a/src/hid/common/hidgl.h
+++ b/src/hid/common/hidgl.h
@@ -59,6 +59,28 @@ hidgl_add_triangle (triangle_buffer *buffer,
   buffer->triangle_count++;
 }
 
+static inline void
+hidgl_add_triangle_3D (triangle_buffer *buffer,
+                       GLfloat x1, GLfloat y1, GLfloat z1,
+                       GLfloat x2, GLfloat y2, GLfloat z2,
+                       GLfloat x3, GLfloat y3, GLfloat z3)
+{
+  if (buffer->triangle_count == TRIANGLE_ARRAY_SIZE) {
+    fprintf (stderr, "hidgl_add_triangle_3D: Attempt to add past end of array. Will now segfault.\n");
+    *(char *)0 = 0;
+  }
+  buffer->triangle_array [buffer->coord_comp_count++] = x1;
+  buffer->triangle_array [buffer->coord_comp_count++] = y1;
+  buffer->triangle_array [buffer->coord_comp_count++] = z1;
+  buffer->triangle_array [buffer->coord_comp_count++] = x2;
+  buffer->triangle_array [buffer->coord_comp_count++] = y2;
+  buffer->triangle_array [buffer->coord_comp_count++] = z2;
+  buffer->triangle_array [buffer->coord_comp_count++] = x3;
+  buffer->triangle_array [buffer->coord_comp_count++] = y3;
+  buffer->triangle_array [buffer->coord_comp_count++] = z3;
+  buffer->triangle_count++;
+}
+
 // void draw_grid ()
 void hidgl_draw_line (int cap, double width, int x1, int y1, int x2, int y2, double scale);
 void hidgl_draw_arc (double width, int vx, int vy, int vrx, int vry, int start_angle, int delta_angle, double scale);
